### PR TITLE
fix(encoding/form): optimize EncodeField and add test cases

### PR DESCRIPTION
**Pull request by demoManito**
#### Description (what this PR does / why we need it):

- remove protoreflect.StringKind convert to string
image

- supplementary encode map test cases

### CODE REVIEW

File: encoding/form/proto_encode.go [hunk 1]
```diff
@@ -137,14 +137,12 @@ func EncodeField(fieldDescriptor protoreflect.FieldDescriptor, value protoreflec
...
 		return string(desc.Name()), nil
-	case protoreflect.StringKind:
-		return value.String(), nil
 	case protoreflect.BytesKind:
 		return base64.URLEncoding.EncodeToString(value.Bytes()), nil
 	case protoreflect.MessageKind, protoreflect.GroupKind:
 		return encodeMessage(fieldDescriptor.Message(), value)
 	default:
-		return fmt.Sprint(value.Interface()), nil
+		return value.String(), nil
```
Comment (demoManito) [inline encoding/form/proto_encode.go:145]: image

Comment (shenqidebaozi) [inline encoding/form/proto_encode.go:145]: Should this be a feature or fix?

Comment (demoManito) [reply encoding/form/proto_encode.go:145]: image

`value.String()` already contains `fmt.Sprint(value.Interface())`, for non-special types, just use `.String()`, optimized by the way.

Comment (shenqidebaozi) [reply encoding/form/proto_encode.go:145]: What I'm talking about is the title issue

Comment (demoManito) [reply encoding/form/proto_encode.go:145]: done @shenqidebaozi

**Pull request merged**